lrlex/src/lib/ctbuilder.rs

```diff
@@ -572,11 +572,16 @@ where
                 }
             }
         };
-
-        // Pretty print it.
-        let out_file = syn::parse_file(&out_tokens.to_string()).unwrap();
-        let outs = prettyplease::unparse(&out_file);
-
+        let outs = out_tokens.to_string();
+        #[cfg(feature = "prettyplease")]
+        let outs = if let Ok(syntax_tree) = syn::parse_str(&outs) {
+            prettyplease::unparse(&syntax_tree)
+        } else {
+            // We failed to parse the source string before pretty printing.
```

Comment:
I'm not really enamored with this

```
let outs = ...;
#[cfg(...)]
let outs = if ... else { outs };
```

Pattern, but like trying to make it `let mut outs` and then going `#[cfg(...)] if Ok(_) { outs = ...  }` lead to clippy warnings because `outs` doesn't need to be mut unless the feature is enabled.

I couldn't think of any nicer way, but would love a better one.

Reply:
I guess using the combinators is preferred here, so I did that in [96549f1](https://github.com/softdevteam/grmtools/pull/531/commits/96549f1e912f80a55c230cb5c471311b39959c00)